src/main/java/org/openrefine/extensions/commons/importer/CommonsImportingController.java

```diff
@@ -0,0 +1,244 @@
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Properties;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.google.refine.ProjectManager;
+import com.google.refine.ProjectMetadata;
+import com.google.refine.RefineServlet;
+import com.google.refine.commands.HttpUtilities;
+import com.google.refine.importers.TabularImportingParserBase;
+import com.google.refine.importers.TabularImportingParserBase.TableDataReader;
+import com.google.refine.importing.ImportingController;
+import com.google.refine.importing.ImportingJob;
+import com.google.refine.importing.ImportingManager;
+import com.google.refine.model.Project;
+import com.google.refine.util.JSONUtilities;
+import com.google.refine.util.ParsingUtilities;
+
+import okhttp3.HttpUrl;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+
+public class CommonsImportingController implements ImportingController {
+    private static final Logger logger = LoggerFactory.getLogger("CommonsImportingController");
+    protected RefineServlet servlet;
+    public static int DEFAULT_PREVIEW_LIMIT = 50;
+    public static int DEFAULT_PROJECT_LIMIT = 0;
+
+    @Override
+    public void init(RefineServlet servlet) {
+        this.servlet = servlet;
+    }
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response)
+        throws ServletException, IOException {
+        HttpUtilities.respond(response, "error", "GET not implemented");
+    }
+
+    /* Handling of http requests between frontend and OpenRefine servlet */
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response)
+        throws ServletException, IOException {
+        if(logger.isDebugEnabled()){
+            logger.debug("doPost Query String::{}", request.getQueryString());
+        }
+        response.setCharacterEncoding("UTF-8");
+        Properties parameters = ParsingUtilities.parseUrlParameters(request);
+
+        String subCommand = parameters.getProperty("subCommand");
+
+        if(logger.isDebugEnabled()){
+            logger.info("doPost::subCommand::{}", subCommand);
+        }
+
+        if ("initialize-parser-ui".equals(subCommand)) {
+            doInitializeParserUI(request, response, parameters);
+        } else if ("parse-preview".equals(subCommand)) {
+            try {
+
+                doParsePreview(request, response, parameters);
+
+            } catch (Exception e) {
+                logger.error("doPost::DatabaseServiceException::{}", e);
```

Comment:
```suggestion
                logger.error("doPost::CommonsExtensionException::{}", e);
```

(this looks like a copy-paste leftover from the database extension)